keps/sig-auth/1205-bound-service-account-tokens/README.md

```diff
@@ -265,53 +258,161 @@ service account token on behalf of pods running on that node. The
+
 This modification could be optionally enabled by providing a command line flag
- to kube apiserver.
+to kube apiserver.
 
-These extended tokens would not expire and continue to be accepted within one year.
-At the same time, the authentication side could monitor whether clients are properly
-reloading tokens by:
+These extended tokens would not expire and continue to be accepted within one
+year. At the same time, the authentication side could monitor whether clients
+are properly reloading tokens by:
 
-  1. Compare the `kubernetes.io/warnafter` field with current time. If current time
-  is after `kubernetes.io/warnafter` field, it implies calling client is not 
-  reloading token regularly.
-  1. Expose metrics to monitor number of legacy and stale token used.
-  1. Add annotation to audit events for legacy and stale tokens including necessary 
-  information to locate problematic client.
- 
-This functionality can be implemented entirely in the kube-apiserver and does not 
-require cooperation by the kubelet beyond the standard functionality implemented 
-as part of [Service Account Token Volumes](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/svcacct-token-volume-source.md).
+1.  Compare the `kubernetes.io/warnafter` field with current time. If current
+    time is after `kubernetes.io/warnafter` field, it implies calling client is
+    not reloading token regularly.
+1.  Expose metrics to monitor number of legacy and stale token used.
+1.  Add annotation to audit events for legacy and stale tokens including
+    necessary information to locate problematic client.
 
-We will direct cluster administrators to adopt bound service account tokens and 
-turn on the command line flag to enable extended token. Fix clients if monitoring or 
-audit report that stale or legacy token are in use. The operator can turn off 
-the command line flag after no stale token is being used.
+### Graduation Criteria
 
+#### Alpha->Beta
 
+All known migration frictions have been fixed.
 
-### Graduation Criteria
+- PodSecurityPolicies that allow secrets but not projected volumes will
+  handicap the provision of token volumes.
+  - Fixed in https://github.com/kubernetes/kubernetes/pull/92006
+- In-cluster clients that don’t reload service account tokens will start
+  failing an hour after deployment.
+  - Fixed in https://github.com/kubernetes/kubernetes/issues/68164
+- Pods running as non root may not access the service account token.
+  - Fixed in https://github.com/kubernetes/kubernetes/pull/89193
 
 #### Beta -> GA Graduation
 
-- TBD
+New `ServiceAccount` admission controller WAI in Beta for >= 1 minor without
+significant issues.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+- **How can this feature be enabled / disabled in a live cluster?**
+
+  - Feature gate name: `BoundServiceAccountVolume`
+  - Components depending on the feature gate: kubelet, kube-apiserver
```

Comment:
only the API server and kube-controller-manager use the `BoundServiceAccountVolume` feature gate

it depends on these features:
* TokenRequest (in API server)
* TokenRequestProjection (in API server and kubelet)

what is the oldest kubelet version that has the `TokenRequestProjection` feature enabled by default?

Reply:
v12.0. https://github.com/kubernetes/kubernetes/pull/67349/commits/92b4e3ca6096070021bf7e42b27a957890e91c67